### PR TITLE
Allow connectivity for non-matching paradim-dim combinations

### PR DESCRIPTION
**Pull request by jzwar**
### Embedded Connectivities
Change code to be less restrictive (no longer required only for mfem export)

# Check list
### Test documentation (delete if not applicable)
* [x]  Build Debug
* [x]  Build Release Mode
* [x]  Executed Tests
* [x]  Build Tests
* [x]  Executed Examples
* [x]  Added new unit test for new feature

### Documentation of changes and theory
* [x]  I have updated the relevant documentations, as far as necessary.

### Style guide compliance
* [x]  Edited `C++` files formatted with clang-format (`10.x.x`+)

### CODE REVIEW

File: tests/io/check_connectivity/connectivity_test.cpp [hunk 1]
```diff
@@ -62,6 +63,18 @@ class ConnectivityCheckSuite : public ::testing::Test {
...
+      Point3D{0., 2., 2.},
+      // Second Patch
+      Point3D{1., 1., 2.}, Point3D{2., 1., 2.}, Point3D{2., 2., 2.},
+      Point3D{1., 2., 2.},
+      // Third Patch
+      Point3D{1., 0., 2.}, Point3D{2., 0., 2.}, Point3D{2., 1., 2.},
+      Point3D{1., 1., 2.}};
+
   // Expected Connectivity
   std::vector<std::array<std::size_t, 4>> expected_connectivity{
```
Comment (j042) [inline tests/io/check_connectivity/connectivity_test.cpp:79]: can you explain this?

Comment (j042) [reply tests/io/check_connectivity/connectivity_test.cpp:79]: `static_cast<std::size_t>(-1)` ?

Comment (jzwar) [reply tests/io/check_connectivity/connectivity_test.cpp:79]: Ah I thought you were referring to the lines above. "negativ" values just mean that there is a boundary, the abs value is the boundary ID

**Review by j042 — APPROVED** [review]
great work

**Pull request merged**